apps/system/js/app_window.js

```diff
@@ -1117,23 +1117,24 @@
 
   AppWindow.prototype._handle_mozbrowserlocationchange =
     function aw__handle_mozbrowserlocationchange(evt) {
-      this.favicons = {};
-      this.webManifestURL = null;
-      this.webManifest = null;
-      this.config.url = evt.detail;
-      this.title = evt.detail;
       if (!this.manifest) {
         this.name = new URL(evt.detail).hostname;
       }
+      this.title = evt.detail;
```

Comment:
Reasonable question. I think the idea is so that we always have some title - and if a page doesnt provide one we fallback to the URL as title. This should be conditional though. I'll move this inside the if block